packages/js/src/plugins/auctionHouseModule/operations/buy.ts

```diff
@@ -0,0 +1,214 @@
+   *
+   * @defaultValue No default value.
+   */
+  tokenAccount?: Option<PublicKey>;
+
+  /**
+   * The buyer's price.
+   *
+   * @defaultValue 0 SOLs or tokens.
+   */
+  price?: SolAmount | SplTokenAmount;
+
+  /**
+   * The number of tokens to bid for.
+   * For an NFT bid it must be 1 token.
+   *
+   * When a Fungible Asset is put on sale.
+   * The buyer can then create a buy order of said assets that is
+   * less than the token_size of the sell order.
+   *
+   * @defaultValue 1 token.
+   */
+  tokens?: SplTokenAmount;
+
+  /**
+   * Prints the bid receipt.
+   * The receipt holds information about the bid,
+   * So it's important to print it if you want to use the `Bid` model
+   *
+   * The receipt printing is skipped for the Auctioneer Auction House
+   * Since it currently doesn't support it.
+   *
+   * @defaultValue `true`
+   */
+  printReceipt?: boolean;
+
+  /**
+   * The address of the bookkeeper wallet responsible for the receipt.
+   *
+   * @defaultValue `metaplex.identity()`
+   */
+  bookkeeper?: Signer;
+
+  /** A set of options to configure how the transaction is sent and confirmed. */
+  confirmOptions?: ConfirmOptions;
+};
+
+/**
+ * @group Operations
+ * @category Outputs
+ */
+export type BuyOutput = {
+  /** Seller trade state account address encoding the listing order. */
+  sellerTradeState: PublicKey;
+
+  /** Buyer trade state account address encoding the bid order. */
+  buyerTradeState: PublicKey;
+
+  /** The buyer address. */
+  buyer: PublicKey;
+
+  /** The seller address. */
+  seller: PublicKey;
+
+  /** The asset's metadata address. */
+  metadata: PublicKey;
+
+  /** The address of the bookkeeper account responsible for the receipt. */
+  bookkeeper: Option<PublicKey>;
+
+  /** The PDA of the receipt account in case it was printed. */
+  receipt: Option<Pda>;
+
+  /** The sale price. */
+  price: SolAmount | SplTokenAmount;
+
+  /** The number of tokens bought. */
+  tokens: SplTokenAmount;
+
+  /** A model that keeps information about the Purchase. */
+  purchase: Purchase;
+
+  /** The blockchain response from sending and confirming the transaction. */
+  response: SendAndConfirmTransactionResponse;
+};
+
+/**
+ * @group Operations
+ * @category Handlers
+ */
+export const buyOperationHandler: OperationHandler<BuyOperation> = {
+  handle: async (operation: BuyOperation, metaplex: Metaplex) => {
+    const { listing, price, buyer, tokenAccount, tokens, authority, ...rest } =
+      operation.input;
+
+    const { bid } = await metaplex
+      .auctionHouse()
+      .bid({
+        authority,
+        price,
+        buyer,
+        tokenAccount,
+        tokens,
+        mintAccount: listing.asset.mint.address,
+        seller: listing.sellerAddress,
+        ...rest,
+      })
+      .run();
+
+    return await metaplex
+      .auctionHouse()
+      .executeSale({
+        bid,
+        listing,
+        ...rest,
+      })
+      .run();
+  },
+};
```

Comment:
This is not what I expected.

The direct sell/buy operations should still expose a transaction builder that delegates to the other two transaction builders. Then, this operation handler simply uses it like most other handlers.